lib/matplotlib/quiver.py

```diff
@@ -372,12 +372,13 @@ def _parse_args(*args):
     X, Y, U, V, C = [None] * 5
     args = list(args)
 
-    # The use of atleast_1d allows for handling scalar arguments while also
-    # keeping masked arrays
-    if len(args) == 3 or len(args) == 5:
+    if len(args) == 2 or len(args) == 4:
+        V = np.atleast_1d(args.pop(-1))
+        U = np.atleast_1d(args.pop(-1))
+    elif len(args) == 3 or len(args) == 5:
```

Comment:
@trpham, I disagree. We need to encapsulate logical chunks of argument validation that typically go together so as to minimize code duplication.  The consolidation I am suggesting here is really minimal--it is ensuring that a given argument is some sort of ndarray but not a matrix.  That's consistent with the name ``_check_array`` (or it could be ``_ensure_array``, etc.)  With regard to matrix checking: *all* of our array-like argument validation should be explicitly blocking (or converting) matrix types because they are known to have odd behavior.  This needs to be embedded in the more general validation and "argument scrubbing" functions.  We have never claimed to support the matrix subclass, and have always known we didn't want to try to do so--but in the evolution of the library, basic functionality has come first, and ever more stringent validation and scrubbing is gradually being added.